1.4.1: test_getuserdir and test_expanduser_user fail during packaging

Hi! I'm packaging python-senf for Arch Linux.

When building and testing 1.4.1 I ran into the following issues with the tests:

```
=================================== FAILURES ===================================
_______________________________ test_getuserdir ________________________________

    def test_getuserdir():
        # type: () -> None

        userdir = _get_userdir()
        assert isinstance(userdir, fsnative)

        with pytest.raises(TypeError):
            _get_userdir(notfsnative(u"foo"))

        if sys.platform == "win32":
            otherdir = _get_userdir(u"foo")
            assert otherdir == os.path.join(os.path.dirname(userdir), u"foo")
        else:
            user = os.path.basename(userdir)
>           assert _get_userdir() == _get_userdir(user)
E           AssertionError: assert '/build' == None
E             -'/build'
E             +None

tests/test_api.py:194: AssertionError
_____________________________ test_expanduser_user _____________________________

    def test_expanduser_user():
        # type: () -> None

        home = _get_userdir()
        user = os.path.basename(home)

>       assert expanduser("~" + user) == home
E       AssertionError: assert '~build' == '/build'
E         - ~build
E         ? ^
E         + /build
E         ? ^

tests/test_api.py:244: AssertionError
=================== 2 failed, 53 passed, 9 skipped in 9.56s ====================
```

Packages are being built in a pristine container, that only gets installed, what the package requires.
The build user home gets reset for each package and the user running the build is called `builduser`, with its `HOME` at `/build` within that container.
Somehow the assumptions in those tests don't seem to be able to cover this use-case (yet).

--- a/tests/test_api.py
+++ b/tests/test_api.py
@@ -177,10 +177,19 @@ def test__get_encoding():
         sys.getfilesystemencoding = orig
 
 
+def _get_real_userdir():
+    # Tries to return the real userdir of the current user
+    # ignore potentially replaced env vars
+    with preserve_environ():
+        environ.pop("HOME", None)
+        userdir = _get_userdir()
+    return userdir or _get_userdir()
+
+
 def test_getuserdir():
     # type: () -> None
 
-    userdir = _get_userdir()
+    userdir = _get_real_userdir()
     assert isinstance(userdir, fsnative)
 
     with pytest.raises(TypeError):
@@ -191,7 +200,7 @@ def test_getuserdir():
         assert otherdir == os.path.join(os.path.dirname(userdir), u"foo")
     else:
         user = os.path.basename(userdir)
-        assert _get_userdir() == _get_userdir(user)
+        assert userdir == _get_userdir(user)
 
     with preserve_environ():
         environ["HOME"] = "bla"
@@ -238,7 +247,7 @@ def test_expanduser_simple():
 def test_expanduser_user():
     # type: () -> None
 
-    home = _get_userdir()
+    home = _get_real_userdir()
     user = os.path.basename(home)
 
     assert expanduser("~" + user) == home

